isis/tests/FindImageOverlapsTests.cpp

```diff
@@ -0,0 +1,120 @@
+#include <iostream>
+#include <QTemporaryFile>
+
+#include "findimageoverlaps.h"
+#include "PolygonTools.h"
+#include "ImageOverlap.h"
+#include "ImageOverlapSet.h"
+#include "Blob.h"
+#include "Pvl.h"
+
+#include <geos/io/WKTReader.h>
+#include <geos/io/WKTWriter.h>
+#include "geos/geom/CoordinateArraySequence.h"
+#include "geos/geom/CoordinateSequence.h"
+#include "geos/geom/LinearRing.h"
+#include "geos/geom/Polygon.h"
+
+#include "Fixtures.h"
+
+#include "gmock/gmock.h"
+
+using namespace Isis;
+
+static QString APP_XML = FileName("$ISISROOT/bin/xml/findimageoverlaps.xml").expanded();
+
+TEST_F(ThreeImageNetwork, FunctionalTestFindImageOverlapDefault) {
+  geos::geom::CoordinateSequence *newLonLatPts1 = new geos::geom::CoordinateArraySequence();
+  newLonLatPts1->add(geos::geom::Coordinate(30, 0));
+  newLonLatPts1->add(geos::geom::Coordinate(30, 10));
+  newLonLatPts1->add(geos::geom::Coordinate(35, 10));
+  newLonLatPts1->add(geos::geom::Coordinate(35, 0));
+  newLonLatPts1->add(geos::geom::Coordinate(30, 0));
+
+  std::vector<geos::geom::Geometry *> *polys1 = new std::vector<geos::geom::Geometry *>;
+  geos::geom::Polygon *poly1 = globalFactory->createPolygon(globalFactory->createLinearRing(newLonLatPts1), nullptr);
+  polys1->push_back(poly1->clone());
+  geos::geom::MultiPolygon *multiPoly1 = globalFactory->createMultiPolygon(polys1);
+
+  geos::geom::CoordinateSequence *newLonLatPts2 = new geos::geom::CoordinateArraySequence();
+  newLonLatPts2->add(geos::geom::Coordinate(31, 1));
+  newLonLatPts2->add(geos::geom::Coordinate(31, 11));
+  newLonLatPts2->add(geos::geom::Coordinate(36, 11));
+  newLonLatPts2->add(geos::geom::Coordinate(36, 1));
+  newLonLatPts2->add(geos::geom::Coordinate(31, 1));
+
+  std::vector<geos::geom::Geometry *> *polys2 = new std::vector<geos::geom::Geometry *>;
+  geos::geom::Polygon *poly2 = globalFactory->createPolygon(globalFactory->createLinearRing(newLonLatPts2), nullptr);
+  polys2->push_back(poly2->clone());
+  geos::geom::MultiPolygon *multiPoly2 = globalFactory->createMultiPolygon(polys2);
+
+  geos::io::WKTWriter *wkt = new geos::io::WKTWriter();
+
+  std::string polyStr1 = wkt->write(multiPoly1);
+  int polyStr1Size = polyStr1.size();
+  std::istringstream dummyStream1(polyStr1);
+  std::string polyStr2 = wkt->write(multiPoly2);
+  int polyStr2Size = polyStr2.size();
+  std::istringstream dummyStream2(polyStr2);
+
+  delete wkt;
```

Comment:
There's a lot of pointers created here and only the WKT writer is being deleted. Is ownership being passed around or all of these just hanging around?

Reply:
It would also be safer to define many of these in a fixture so that they get clean-up if something errors.